src/Cake.Common.Tests/Unit/Solution/Project/ProjectParserTests.cs

```diff
@@ -0,0 +1,193 @@
+
+            // Then
+            Assert.Equal(result.TargetFrameworkVersion, "v4.5");
+        }
+
+        [Fact]
+        public void Should_Parse_Framework_Profile()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal("Profile111", result.TargetFrameworkProfile);
+        }
+
+        [Fact]
+        public void Should_Return_Null_When_Profile_Not_Defined()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.ParseIncomplete();
+
+            // Then
+            Assert.NotNull(result);
+            Assert.Null(result.TargetFrameworkProfile);
+        }
+
+        [Fact]
+        public void Should_Parse_Assembly_Name()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(result.AssemblyName, "Cake.Common");
+        }
+
+        [Fact]
+        public void Should_Parse_Namespace()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(result.RootNameSpace, "Cake.Common");
+        }
+
+        [Fact]
+        public void Should_Parse_Output_Type()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(result.OutputType, "Library");
+        }
+
+        [Fact]
+        public void Should_Parse_Configuration()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(result.Configuration, "Debug");
+        }
+
+        [Fact]
+        public void Should_Parse_Platform()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(result.Platform, "AnyCPU");
+        }
+
+        [Fact]
+        public void Should_Return_Correct_File_Count()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            Assert.Equal(2, result.Files.Count);
+        }
+
+        [Fact]
+        public void Should_Return_Valid_Guid()
+        {
+            // Given
+            var fixture = new ProjectParserFixture();
+
+            // When
+            var result = fixture.Parse();
+
+            // Then
+            var g = new Guid();
```

Comment:
Inital value never used this can be
`Guid g;`
also a minor naming like `projectGuid` would likely be more clear than `g`